react/forms.rst

```diff
@@ -0,0 +1,206 @@
+.. _forms-label:
+
+========================
+Use Forms To Add An Item
+========================
+
+Add The Form
+============
+
+To be able to add Faq items to the list we'll start by adding an add form:
+
+::
+
+    render() {
+      return (
+        <div>
+          <ul>
+            {this.state.faq.map((item, index) => (
+              <FaqItem
+                question={item.question}
+                answer={item.answer}
+                index={index}
+                onDelete={this.onDelete}
+              />
+            ))}
+          </ul>
+          <form>
+            <label>
+              Question: <input name="question" type="text" />
+            </label>
+            <label>
+              Answer: <textarea name="answer" />
+            </label>
+            <input type="submit" value="Add" />
+          </form>
+        </div>
+      );
+    }
+
+Manage Field Values In The State
+================================
+
+To manage the values of the fields in the form we will use the state. Add a
+question and answer value to the state which contains the values of the inputs.
+Add :file:`onChange` handlers to the input and textarea which will change the
+values in the state when the input changes. This pattern is called controlled
+inputs.
+
+..  admonition:: Solution
+    :class: toggle
+
+    ::
```

Comment:
```
    .. code-block::
        :language: js
```

Reply:
same here, line length and one sentence per line